crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -348,6 +350,16 @@ impl<'t> TextEdit<'t> {
         self.min_size = min_size;
         self
     }
+
+    /// Set the return key combination.
+    ///
+    /// This combination will cause a newline on multiline,
+    /// whereas on singleline it will cause the widget to lose focus.
+    #[inline]
+    pub fn return_key(mut self, return_key: KeyboardShortcut) -> Self {
```

Comment:
I think this should be called `newline_key` instead

Reply:
that was also my first idea, but in the single line mode there is no newline involved at all, so it didn't feel right